Source/player.cpp

```diff
@@ -224,22 +225,22 @@ int PlayerStruct::GetMaximumAttributeValue(CharacterAttribute attribute) const
 	return MaxStats[static_cast<std::size_t>(_pClass)][static_cast<std::size_t>(attribute)];
 }
 
-void PlayerStruct::UpdateTargetPosition()
+SDL_Point PlayerStruct::GetTargetPosition() const

```

Comment:
Should we make this:
```suggestion
const SDL_Point& PlayerStruct::GetTargetPosition() const
```

Reply:
The general guidance is that it's better to return small types (< 4 ints) by value, as the cost of copying for these is much smaller than the cost of memory indirection

Reply:
I'm unsure how mem management works with something like that?

Reply:
> The general rule of thumb is that it's better to return small types (e.g <= 16 bytes) by value, as the cost of copying for these is much smaller than the cost of memory indirection

That was my thinking when I went with the current implementation :D Well by small I was thinking 8 bytes is defiantly small on x86_64 :D

Reply:
Note that in this case guaranteed RVO (return-value-optimization) applies, so the size of the return type doesn't even matter.

You're right that returning by reference here would crash upon first use (pointer to stack variable).